### PR TITLE
:sparkles: Feat/product 2023 04

### PR DESCRIPTION
**Pull request by guillaumeprevost76**

### CODE REVIEW

File: saagieapi/jobs/jobs.py [hunk 1]
```diff
@@ -1627,3 +1823,107 @@ def count_instances_by_status(self, job_id):
         return self.saagie_api.client.execute(
             query=gql(GQL_COUNT_INSTANCES_BY_SELECTOR), variable_values={"jobId": job_id}
         )
+
+    def count_instances_by_date(self, job_id: str, date_before: str):
+        """Count job instances by date
```
Comment (Qiwei08) [inline saagieapi/jobs/jobs.py:1828]: Maybe adjust the description to: `Count removable job instance by date`

Comment (guillaumeprevost76) [reply saagieapi/jobs/jobs.py:1828]: This function counts all instances of a job by date, not only the removable instances. So I think the description is correct.

File: saagieapi/jobs/gql_queries.py [hunk 2]
```diff
@@ -388,6 +388,100 @@
...
+            emails
+            loginEmails{
+                login
+                email
+            }
+            statusList
+        }
+        instances(limit: $instancesLimit){
+            id
+            status
```
[the remark this answers is not in the document]
Comment (guillaumeprevost76) [reply saagieapi/jobs/gql_queries.py:408]: You're right, I remove this field (and by the way it's already accessible in history.currentStatus.status)

File: saagieapi/jobs/gql_queries.py [hunk 3]
```diff
@@ -388,6 +388,100 @@
...
+            }
+            startTime
+            endTime
+            version {
+                number
+                releaseNote
+                runtimeVersion
+                commandLine
+                isMajor
+                doesUseGPU
```
[the remark this answers is not in the document]
Comment (guillaumeprevost76) [reply saagieapi/jobs/gql_queries.py:424]: You're rigth, I remove the field.

File: saagieapi/jobs/jobs.py (hunk 1 above)
Comment (guillaumeprevost76) [reply saagieapi/jobs/jobs.py]: I changed the description and also the name of the function to : count_deletable_instances_by_date

**Review by Qiwei08 — APPROVED** [review]

**Pull request merged**